etc/buildsys/base.mk

```
@@ -29,3 +29,13 @@ else
   include $(BUILDSYSDIR)/rules.mk
 endif
 
+ifeq ($(FAIL_ON_WARNING),1)
+  ifneq ($(WARN_TARGETS),)
+all: $(WARN_TARGETS) fail_on_warning
```

```suggestion
    WARN_TARGETS += fail_on_warning
```
Shouldn't that be enough? Messing with the all target can get you in trouble.

---

No, that doesn't work, because then `fail_on_warning` is added to `WARN_TARGETS` after adding the dependency of `all` on `WARN_TARGETS`, thus it will not contain `fail_on_warning`.

---

But we can simplify it to `all: fail_on_warning`, as this will just add the dependency.

---

Hmm, no, I'd like touching the all target if at all possible. I'll have a deeper look later tonight or tomorrow.

---

But why? This only adds the prerequisite to the target all, it doesn't change it in any other way.